crates/superposition/src/main.rs

```diff
@@ -83,6 +84,25 @@ async fn main() -> Result<()> {
             .split(',')
             .map(String::from)
             .collect::<HashSet<String>>();
+    let mandatory_dimensions: Map<String, Value> =
+        get_from_env_unsafe::<String>("MANDATORY_DIMENSIONS")
+            .expect("MANDATORY_DIMENSIONS is not set")
+            .split(';')
+            .filter_map(|ele| {
+                let arr: Vec<&str> = ele.split(':').collect();
+                if arr.len() == 2 {
+                    let key = arr[0].to_string();
+                    let values =
+                        arr[1].split(',').map(String::from).collect::<Vec<String>>();
+                    Some((
+                        key,
+                        Value::Array(values.into_iter().map(Value::String).collect()),
```

Comment:
this would then simply become
```suggestion
                        Value::Array(values.collect()),
```